### PR TITLE
Add method to upsert one document in collections

### PR DESCRIPTION
**Pull request by pierremalletneo9**

### CODE REVIEW

File: test/tests/index.ts [hunk 1]
```diff
@@ -54,5 +55,73 @@ test('[CRUD] Insert one and find it', async (t: Assertions) => {
 	await mongoClient.dropCollection();
 });
 
+test('[CRUD] Find one and update', async (t: Assertions) => {
+	const mongoClient = new MongoClient('test-' + Date.now(), SampleType, SampleTypeListing);
+	const size = await mongoClient.count();
+
+	t.true(size === 0);
+
+	const randomInt = Math.ceil(Math.random() * Number.MAX_SAFE_INTEGER);
```
[the remark this answers is not in the document]
Comment (pierremalletneo9) [reply test/tests/index.ts]: no idea, it was like that in other tests cases =)

File: test/tests/index.ts [hunk 2]
```diff
@@ -54,5 +55,73 @@ test('[CRUD] Insert one and find it', async (t: Assertions) => {
...
+	}, 'userId1');
+
+	const updateQuery = { $set: { field2Number: randomInt2 }};
+
+	const founded = await mongoClient.findOneAndUpdate({
+			field1String: 'string1',
+			field2Number: randomInt,
+		}, updateQuery, 'userId');
+
+	t.is(founded.field2Number, updateQuery.$set.field2Number, 'Element has been updated');
```
[the remark this answers is not in the document]
Comment (pierremalletneo9) [reply test/tests/index.ts]: it ensure it does not create another if a document matches the query. Should i remove it ?

[the remark this answers is not in the document]
Comment (pierremalletneo9) [reply test/tests/index.ts]: ok i see the point, i change the used method to findOneAndUpsert =)

**Comment by pierremalletneo9** [comment]
PR updated

File: test/tests/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (BenjD90) [reply test/tests/index.ts]: ok, use const please ;)

[the remark this answers is not in the document]
Comment (pierremalletneo9) [reply test/tests/index.ts]: It is done in the updated PR. Sorry i made a force push so all link are wrecked

File: test/tests/index.ts [hunk 3]
```diff
@@ -32,10 +33,10 @@ test('[CRUD] Insert one and find it', async (t: Assertions) => {
 
 	t.true(size === 0);
```
Comment (BenjD90) [inline test/tests/index.ts]: add assert message please ;)

**Pull request merged**